### PR TITLE
First crack at updating the rules of engagement for Contributors

### PR DESCRIPTION
**Pull request by drllau**
Also added in private license (for co-contributors) and waiver (if want to grant community some extra trial beyond 1 month

### CODE REVIEW

File: CONTRIBUTING.md
Comment (zspencer) [inline CONTRIBUTING.md]: I really like principle 2; but am not entirely sure I would be comfortable adopting 1 and 3 as a maintainer. I say many things I don't really care to share with my mom; but if she found out it wouldn't be that big a deal. I also worry about how it appeals to femininity acting as the 'voice of civility' when really, us dudes can and should be able to be thoughtful and considerate as an expression of our masculinity. I also think that the choice of license belongs to the maintainer(s); not to the contributors. See https://github.com/EthicalSource/ethicalsource.dev/pull/49#discussion_r392701879 for commentary by a legal professional who works for RedHat.

At present, I think I'd like to leave the principles chunk out (or make it in a separate patch so we can workshop it with the other @zinc-collective/compensated-maintainers).

We did some initial work setting up ground rules in a commit that slipped in between when you forked initially and now that may serve these purposes. See https://github.com/zinc-collective/compensated/pull/40

File: CONTRIBUTING.md
Comment (zspencer) [inline CONTRIBUTING.md:16]: I like the framing of these quite a bit! I'm trying to think about how we can shrink it down and make the language tighter and more approachable; and I wonder if the appropriate place would be a FAQ section  underneath the [legal heading](https://github.com/zinc-collective/compensated/blob/0.X/CONTRIBUTING.md#legal) in the current verison.

File: CONTRIBUTING.md
Comment (zspencer) [inline CONTRIBUTING.md:34]: I think this is a great point that the current CONTRIBUTING.md is missing.

File: CONTRIBUTING.md
Comment (zspencer) [inline CONTRIBUTING.md]: This is a damn fine point that we probably want to add to the roadmap or create other structures for internationalization. Maybe we can rely on something like [CrowdTranslate](https://github.com/transparentclassroom/crowdtranslate) to do that?

File: CONTRIBUTING.md
Comment (zspencer) [inline CONTRIBUTING.md:46]: Another great point. I wonder if this belongs more on the landing page under a title like "Why are you charging? Shouldn't open source be free?" or something

**Review by zspencer — CHANGES_REQUESTED** [review]
I'm going to submit a patch to your branch that fixes the merge conflicts while keeping your insights. 

Once you merge it; can you mention the @zinc-collective/compensated-maintainers group so that someone can come through and merge it upstream?

Before we can merge your contributions; would you be able to follow the instructions for assigning ownership to Zinc Collective? https://github.com/zinc-collective/compensated/blob/0.X/CONTRIBUTING.md#assignment-of-ownership-to-zinc-collective

File: CONTRIBUTING.md
Comment (drllau) [reply CONTRIBUTING.md]: Since **you are the originator** you chose the license and subsequently insist on all co-contributor(s) to assign rights to you under License Zero which in my opinion is legally unstable (eg is displaying ads commercial?). I'm pointing out that if people write ORIGINAL code which is NOT core, then the only way to be compatible is a CC BY-ND and I'd suspect you'd have to compensate them with:
1. a license to 0.x.* ...
2. waiver for all processing fees until 1.x
3. otherwise just give them $$$ and treat as work-for-hire
4. a non-compete in the sense that they are not allowed to fork the project (the direct implication of Prosperity License) without your express permission

The reason why the other GPL/OSI licenses hold up is that people understand the dynamics over time. Are you expecting to write everything yourself (in which case nobody cares about contributing) or do you want things to be fair and upfront? In which case you need to stipulate the values and principles, debate and get buyin, and stick to them (the forming,storming,norming phase of community). I could replace item#1 with **_Zee aspires to be Linus v2.0 and be annointed benevolent dictator for life_** (sorry as Kiwi, I call things as they are :-D) in which case the situation is totally clear if not popular. So there is the choice between allowing creative programming exploration vs quality control when it comes to squishing phishing and fake invoicing.

item #3 was something to just say "don't be a jerk" which you go into detail much later. Can tweak phrasing to fit.

> follow the instructions for assigning ownership to Zinc Collective?

**Only for the documentation** as I seriously doubt I can use a US stripe processor in my lifetime and I've got other working elsewhere on [payments](https://docs.openpayments.dev/overview) which for the record is a declaration of interest but should NOT be a conflict since my focus is more international. Let me see if there's a common release template somewhere after some [legal research](https://www.researchgate.net/publication/272688529_Comparative_Analysis_of_copyright_assignment_and_licence_formalities_for_Open_Source_Contributor_Agreements).

Comment (zspencer) [reply CONTRIBUTING.md]: Thanks! This is helpful!

Re: Item 1 - For now, I think we _do_ want the license to be decided by the project maintainers; just like how in (from what I can tell) all other open-source projects the license is decided by the maintainers of the project. At present, there is [one person](https://github.com/orgs/zinc-collective/teams/compensated-maintainers) in the compensated maintainers group in Zinc (me). I am more than happy to welcome additional maintainers who regularly offer substantive contributions (either in intellectual property or in conversation, such as this one now). At present, there are two members of the coop who would qualify as additional maintainers, and I've reached out to both to see if they'd like to offset the BDFN (Benevolent Dictator For Now) status of the project.

Re: Item 3 - I believe item 3 is covered by the [code of conduct](https://github.com/zinc-collective/compensated/blob/0.X/CODE_OF_CONDUCT.md); which is item 1 of our [contributor](https://github.com/zinc-collective/compensated/blob/0.X/CONTRIBUTING.md#contributor-ground-rules) and [maintainer ground rules](https://github.com/zinc-collective/compensated/blob/0.X/CONTRIBUTING.md#maintainer-ground-rules).

Re: Assignment of copyright only for documentation - That's reasonable! I believe it becomes incumbent upon the contributor (once they've agreed to assign copyright) to submit only the patches they are comfortable assigning ownership with. I don't know that it's possible for me to keep track of what each contributors preferences are for what they decide to keep and what they decide to contribute back.

Re: Adding other payment processors - The way I've structured the project is such that it should be possible (but not immediately obvious) for people to extend the currently supported payment processors with their own. If you would like to maintain authorship of the payment processor code (and therefore decide on your own preferred license) you will want to create a new repository, name it something like `compensated-<payment-processor>` and follow the pattern outlined in [the stripe event parser](https://github.com/zinc-collective/compensated/blob/0.X/compensated-ruby/lib/compensated/stripe/event_parser.rb). That said, this is very much alpha software so I expect the developer-facing APIs to change through the `0.X` branch and solidify before we reach `1.X`; so there may be some churn. I'll do my best to mitigate that and contribute updates when we do break things when possible.

Comment (zspencer) [reply CONTRIBUTING.md]: P.s. I wrote a ticket purely for discussing how to license Compensated going forward while taking into account your concerns - https://github.com/zinc-collective/compensated/issues/43

Comment (drllau) [reply CONTRIBUTING.md]: Thanks @zspencer (benevolent dictator for now), what you wrote makes sense as I've always been concerned about sustainability (when {creators} n {maintainers} = {} nullset ). The way that I read how the Prosperity License interact with the CC-BY-ND is that the documentation can be redistributed as is but only for versions 0.x (as per your observation in another thread). But this would lapse (sorry, can't think of the actual legal word for now) for 1.x (another new codebase). This is logical since if you are including sample code or documentation, they need to be revised/confirmed for the next release so the cryptoeconomics (aka game theory) says that when waiver for 0.x expires, they need to pay a annual fee to continue using deprecated codebase, they'd do the calculus of whether their one-off mantainance effort of updating their API+docs <= expected lifetime cost of version 1.x. If their payment channel is stable, then they'd be reluctant to switch implies having an rational economic incentive to update docs to v.1.x and continue gratis usage. Should be a no-brainer. 

I've also found a Harmony contributor agreement for corporate entities that want to donate code which is the other extreme case of simple email "I assign copyright". But github doesn't seem to allow uploading (below this message) so I put here (./core/LICENSE/Zinc%20Collective%20LLC-Entity%20Assigment%20of%20Rights%20by%20Corporates.pdf)
So I think you've covered the spectrum from casual commentor (just email maintainers), to the expected use case (hackavista + SME)  (a formal esig and affirmed in CREDITS.md) to the ultrarare generous corporate.

File: CONTRIBUTING.md
Comment (drllau) [reply CONTRIBUTING.md]: Given the legal headaches with GDPR, let's keep internationalisation off 0.x for now

File: CONTRIBUTING.md
Comment (drllau) [reply CONTRIBUTING.md:46]: I have a very specific order base on my years of [pyramid pitching](https://venturebeat.com/2012/09/10/an-entrepreneurs-most-powerful-tool-the-pyramid-pitch/) to investors 

1. Why?
2. How to ....
3. Cost/benefits or tradeoffs (well not a financial pitch for DevOps but you get the idea)
4. Who
5. Hilights or take away points (optional)
8 . Changes, now or anticipated.
9 . RTFM ... references
People TL:DR; get by with WHY and HOW. You get more differentiation at WHO so once I get text right, add internal jumppoints

File: CONTRIBUTING.md
Comment (drllau) [reply CONTRIBUTING.md:16]: I assign all rights, including copyright, in https://github.com/zinc-collective/compensated/*.[md,html,]? and /core to Zinc Collective LLC.

File: CONTRIBUTING.md
Comment (drllau) [reply CONTRIBUTING.md:46]: You can remember my general order with the mnemomic I came up with **eye1, how2, for3, who4**,  + *hiv5 mut8 and ref9* with zero left for indexing. You she how it goes why one [read] how to for free [benefiting] who for, high five seeing mutate [and] refine sear[ch] row.

**Review by drllau — COMMENTED** [review]
Still a little confused as to why this and another pull request can coexist but if merged, then can delete the old branches ... there's a couple files (images) I need to upload but it is refusing write permission for now. Onwards.

**Comment by zspencer** [comment]
Yea, github... does not really play nice with Git's federated source model. There's probably changes I need to make to how I use it as well (squash merge and rebase merge appear to make things weird).

I've merged what I could of [your changes](https://github.com/zinc-collective/compensated/pull/42) and if you want to start a new branch off of zinc-collective/0.X to add the images you were looking for that would probably work?

File: CONTRIBUTING.md
Comment (zspencer) [reply CONTRIBUTING.md]: Thanks!

**Pull request closed without merging**
